### ISSUE TITLE
** sometimes matches dotfiles

### ISSUE THREAD

**Issue by MartinKolarik** [micromatch/micromatch#63]
From #62:

``` js
mm.isMatch('/aaa/bbb/.git', '/aaa/bbb/**'); // => true; should be false
mm.isMatch('aaa/bbb/.git', 'aaa/bbb/**'); // => true; should be false
mm.isMatch('/aaa/bbb/ccc/.git', '/aaa/bbb/**'); // => false; correct
```

**Comment by tunnckoCore** [comment]
And

```js
var path = require('path')
var mm = require('micromatch')

console.log(mm.isMatch(path.resolve('.git'), path.resolve('**'))) // => true
console.log(mm.isMatch(path.resolve('.git'), path.resolve('**'), {dot: true})) // => true
console.log(mm.isMatch('.git', '**')) // => false
console.log(mm.isMatch('.git', '**', {dot: true})) // => true
```

I'm not sure totally, but I think it make sense first one also to be `false`. It may be bash spec behaving, but it not make sense to get dot files/dirs  when you don't want. And it's normal user to not expect them when he read `dot is false by default` in first time.

**Comment by MartinKolarik** [comment]
@tunnckoCore the first one is exactly what this issue is about - it should be `false`.

**Comment by wtgtybhertgeghgtwtg** [comment]
This was resolved in `nanomatch`, but isn't that only used in `micromatch@3.0.0`?  I think this is still a problem in the current version.

**Comment by jonschlinkert** [comment]
> I think this is still a problem in the current version.

thanks this was automatically closed. I'll reopen until the dev branch is merged

**Comment by tunnckoCore** [comment]
> I'll reopen until the dev branch is merged

we all expecting it :)) good luck to get to it soon!

**Comment by jonschlinkert** [comment]
I believe this was resolved in 3.0.0 but please reopen if you still experience the issue
